client/src/api/v3/boost.ts

```diff
@@ -1,28 +1,121 @@
+ * Send configuration status over MQTT on topic 'boost/configs/action'
  *
+ * @param actionType represents whether the config is being uploaded or deleted
  * @param type the type of the configuration being sent
- * @param configFile file containing the configuration content
+ * @param name name of the config file
+ * @param configContent configuration content
+ */
+function sendConfig(
+  actionType: payloadAction,
+  type: ConfigT,
+  name: string,
+  configContent: string | null,
+) {
+  const channel = 'send-config';
+  const payload = {
+    action: actionType,
+    configType: type,
+    fileName: name,
+    content: configContent,
+  };
+  // To check things work without needing the server started
+  console.log(payload);
+  emit(channel, JSON.stringify(payload));
+}
+
+const configTypeToFileSuffix: { [K in ConfigT]: string } = {
+  rider: '_rider.json',
+  bike: '_bike.json',
+  track: '_track.json',
+  powerPlan: '_powerPlan.json',
+};
+
+/**
+ * Split the given configurations object into inidvidual configs and then send them over MQTT
+ *
+ * @param configs dictionary containign the 4 config types (`powerPlan`, `rider`, `track` and `bike`)
+ * @param fileName name of the file that contained all the configs
+ */
+function uploadMultipleConfigs(configs: ConfigBundleT, fileName: string) {
+  // For each config, send the config content over MQTT
+  Object.entries(configs).forEach((configEntry) => {
+    const configType = configEntry[0] as ConfigT;
+    const config = ConfigObjT.check(configEntry[1]);
+
+    // Since this config was uploaded as a bundle give it a different file name to differnetiate it's config type.
+    const file = fileName.replace('.json', configTypeToFileSuffix[configType]);
+
+    sendConfig('upload', configType, file, JSON.stringify(config));
+  });
+
+  toast.success(`Uploaded configs in ${fileName}`);
+}
+
+/**
+ * Read content from the given file and send it on `boost/configs/action` over MQTT.
+ * If the content contains more than one config (i.e. `type` is 'all'), the content is
+ * split into the different configurations before sending.
+ *
+ * @param type the type of the configuration
+ * @param configFile file containing content of the configuration
+ * @param displayErr function to display error if uploaded config is not correct
  */
 export default function uploadConfig(
-  type: BoostConfigType,
+  type: FileConfigT,
   configFile: File,
+  displayErr: (message: string) => void,
 ) {
-  const topic = 'send-config';
   const reader = new FileReader();
 
   // Called when FileReader has completed reading a file
   reader.onload = () => {
-    console.log(reader.result);
-    const payload = {
-      action: 'upload',
-      configType: type,
-      content: reader.result,
-    };
-    emit(topic, JSON.stringify(payload));
+    const fileContent = reader.result as string;
+    const configContent = JSON.parse(fileContent);
+
+    if (!isCorrectContentType(configContent, fileConfigTypeToRuntype[type])) {
```

Comment:
This could be replaced directly with a `try-catch` block. I feel that would be more understandable and follow a more intuitive logic flow (i.e. do stuff, and if an exception is raised, handle it).

Reply:
Completely understandable, funnily though I purposely removed it because I don't like `try-catch` block when there's more than 2 or 3 lines of code that needs to be executed in the `try` section because of the extra (necessary imo) indentation. As for the understandability concern, I kind of feel the function name kind of handles it. Again, this was solely based on my personal preference, let me know if you strongly think it should be the other way.

Reply:
Similar to before, I don't think it reduces the indentation level overall. Also, it might just be a preference thing so I'm happy to wait and get other's inputs on this.

Reply:
Another point: using this function does not allow us to leverage run type's type guards, so we would need to use explicit type coercion.